packages/wrangler/src/__tests__/ai.test.ts

```diff
@@ -0,0 +1,135 @@
+import { rest } from "msw";
+import { endEventLoop } from "./helpers/end-event-loop";
+import { mockAccountId, mockApiToken } from "./helpers/mock-account-id";
+import { mockConsoleMethods } from "./helpers/mock-console";
+import { clearDialogs } from "./helpers/mock-dialogs";
+import { useMockIsTTY } from "./helpers/mock-istty";
+import { createFetchResult, msw } from "./helpers/msw";
+import { runInTempDir } from "./helpers/run-in-tmp";
+import { runWrangler } from "./helpers/run-wrangler";
+
+describe("ai help", () => {
+	const std = mockConsoleMethods();
+	runInTempDir();
+
+	it("should show help when no argument is passed", async () => {
+		await runWrangler("ai");
+		await endEventLoop();
+
+		expect(std.out).toMatchInlineSnapshot(`
+		"wrangler ai
+
+		🤖 Interact with AI models
+
+		Commands:
+		  wrangler ai models  List catalog models
+
+		Flags:
+		  -j, --experimental-json-config  Experimental: Support wrangler.json  [boolean]
+		  -c, --config                    Path to .toml configuration file  [string]
+		  -e, --env                       Environment to use for operations and .env files  [string]
+		  -h, --help                      Show help  [boolean]
+		  -v, --version                   Show version number  [boolean]"
+	`);
+	});
+
+	it("should show help when an invalid argument is passed", async () => {
+		await expect(() => runWrangler("ai asdf")).rejects.toThrow(
+			"Unknown argument: asdf"
+		);
+
+		expect(std.err).toMatchInlineSnapshot(`
+		"[31mX [41;31m[[41;97mERROR[41;31m][0m [1mUnknown argument: asdf[0m
+
+		"
+	`);
+		expect(std.out).toMatchInlineSnapshot(`
+		"
+		wrangler ai
+
+		🤖 Interact with AI models
+
+		Commands:
+		  wrangler ai models  List catalog models
+
+		Flags:
+		  -j, --experimental-json-config  Experimental: Support wrangler.json  [boolean]
+		  -c, --config                    Path to .toml configuration file  [string]
+		  -e, --env                       Environment to use for operations and .env files  [string]
+		  -h, --help                      Show help  [boolean]
+		  -v, --version                   Show version number  [boolean]"
+	`);
+	});
+});
+
+describe("ai commands", () => {
+	mockAccountId();
+	mockApiToken();
+	runInTempDir();
+	const { setIsTTY } = useMockIsTTY();
+
+	const std = mockConsoleMethods();
+
+	beforeEach(() => {
+		// @ts-expect-error we're using a very simple setTimeout mock here
+		jest.spyOn(global, "setTimeout").mockImplementation((fn, _period) => {
+			setImmediate(fn);
+		});
+		setIsTTY(true);
+	});
```

Comment:
Why is this needed?

Reply:
It was present on other tests on which I based this, I assumed setTimeout() was being used somewhere in the code that the tests called. I'll remove it.

Reply:
Removed.